spyderlib/widgets/file_switcher.py

```diff
@@ -0,0 +1,573 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright © 2013-2015 The Spyder Development Team
+# Copyright © 2015 Daniel Manson (@d1manson)
+# Licensed under the terms of the MIT License
+# (see spyderlib/__init__.py for details)
+
+from __future__ import print_function
+
+import os
+import os.path as osp
+
+from spyderlib.qt.QtGui import (QDialog, QHBoxLayout, QIcon, QLabel, QLineEdit,
+                                QListWidget, QListWidgetItem, QRegExpValidator,
+                                QTextCursor, QVBoxLayout)
+from spyderlib.qt.QtCore import Signal, QEvent, QObject, QRegExp, QSize, Qt
+
+# Local imports
+from spyderlib.config.base import _
+from spyderlib.py3compat import iteritems, to_text_string
+from spyderlib.utils import icon_manager as ima
+from spyderlib.utils.stringmatching import get_search_scores
+from spyderlib.widgets.helperwidgets import HelperToolButton, HTMLDelegate
+
+
+# --- Python Outline explorer helpers
+def process_python_symbol_data(oedata):
+    """Returns a list with line number, definition name, fold and token."""
+    symbol_list = []
+    for key in oedata:
+        val = oedata[key]
+        if val and val != 'found_cell_separators':
+            if val.is_class_or_function():
```

Comment:
When the file has cells I get the following error here:
`AttributeError: 'bool' object has no attribute 'is_class_or_function'`
If you simply create an empty file and add a `#%%` line you get this error.